substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVLineRecord.java

```diff
@@ -124,13 +124,19 @@ public String toString() {
 
         private final ArrayList<LineEntry> lineEntries = new ArrayList<>(DEFAULT_LINE_ENTRY_COUNT);
         private final int fileId;
+        private int previousLine;
 
         FileBlock(int fileId) {
             this.fileId = fileId;
+            this.previousLine = -2;
         }
 
         void addEntry(LineEntry le) {
-            lineEntries.add(le);
+            /* Only add an entry if the line number has changed. */
+            if (le.lineAndFlags != previousLine) {
+                lineEntries.add(le);
+                previousLine = le.lineAndFlags;
+            }
```

Comment:
I'd remove the `previousLine` field and use this instead:
```suggestion
            if (lineEntries.isEmpty() || lineEntries.get(lineEntries.size() - 1).lineAndFlags != le.lineAndFlags) {
                lineEntries.add(le);
            }
```

Otherwise, we need a comment explaining the choice of `-2` as the initial value. 😉

Reply:
Would you be okay with a clarifying comment?  I think the code is easier to read if I leave it as is.  Something like
```
/* Use -2 because it will never appears as a source line. -1 is used as 'no line. */
private static final int NO_PREVIOUS_LINE = -2;
```

Reply:
Well, that would be better, although technically `-2` is possible as this is a value of `lineAndFlags` (it's just that we currently always set high bits to `0`).

What do you think about changing the `previousLine` type to `Integer` and initializing the field to `null`?

Reply:
I'd rather avoid the potential object allocation (even if a good compiler could remove it).  -2 is fairly unlikely, as the line number would have to be 16777214.  I could modify the assert so Graal complains if LineEntry of 16777214 is ever created.

Reply:
Yes, an assertion would suffice, and I'd add it here rather than in `LineEntry`:
```java
        void addEntry(LineEntry le) {
            assert le.lineAndFlags != NO_PREVIOUS_LINE;
            /* Only add an entry if the line number has changed. */
            ...
```

Reply:
Agreed - done.